interfaces/repo.go

```diff
@@ -449,28 +449,73 @@ func (r *Repository) PlugConnections(snapName, plugName string) []*Slot {
 	return result
 }
 
-// Support for sort.Interface
-
-type byPlugSnapAndName []*Plug
+// SlotConnections returns all of the plugs that are connected a given slot.
+func (r *Repository) SlotConnections(snapName, slotName string) []*Plug {
```

Comment:
I've fixed everything else. I see why this is controversial, let's talk as suggested and find a better way to do this.